catalog/app/app.tsx

```diff
@@ -1,16 +1,26 @@
 /* app.tsx - application entry point */
 /* eslint-disable import/first */
+/* eslint-disable import/order */
+
+import { createBrowserHistory as createHistory } from 'history'
+
+// initialize config from window.QUILT_CATALOG_CONFIG
+import cfg from 'constants/config'
+import * as Sentry from 'utils/Sentry'
```

Comment:
the point is to initialize it before other modules can import it synchronously (`import * as Sentry from '@sentry/react'`) and call its methods e.g. `Sentry.captureException`. i've tried to clarify this a little bit in comments (and also removed import/order exception bc it doesn't seem to be required anymore)

Reply:
Maybe, I miss something, but I think `captureException` should work even if you init sentry after all imports.

```
// app.tsx
import moduleA;
import allModules;
import Sentry;
Sentry.init()

// moduleA;
import Sentry;
export default () => {
  Sentry.captureException();
}
```

So, it makes sense to put sentry above to catch exceptions in self-invoking functions

Reply:
I see that you put Sentry init beneath external libraries imports.
And above our modules. I think, it ok because, our modules have some IIFE with higher probability

Reply:
> Maybe, I miss something, but I think captureException should work even if you init sentry after all imports.

yes, but only if we don't have side-effects in a module body

> So, it makes sense to put sentry initialization above to catch exceptions in self-invoking functions

not necessarily self-invoking functions, just any effectful code in module body

> I see that you put Sentry init beneath external libraries imports.
And above our modules. I think, it ok because, our modules have some IIFE with higher probability

yeah, we probably should not expect catching anything in 3-party modules, but some of our modules may contain some side-effects, especially now when the config can be imported synchronously